microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/faulttolerance/java/MicroProfileFaultToleranceASTValidator.java

```diff
@@ -284,11 +284,21 @@ private void validateRetryAnnotation(NormalAnnotation annotation)
 					? jitterExpr.resolveConstantExpressionValue()
 					: null;
 
-			int delayNum = delayConstant != null ? (int) delayConstant : -1;
-			int maxDurationNum = maxDurationConstant != null
-					? (int) maxDurationConstant
-					: -1;
-			int jitterNum = jitterConstant != null ? (int) jitterConstant : 0;
+			int delayNum = -1;
+			int maxDurationNum = -1;
+			int jitterNum = 0;
+
+			if (delayConstant instanceof Integer) {

```

Comment:
Similar to the ternary assignment before, you can add the instance check as a condition:
```java
int delayNum = (delayConstant != null && delayConstant instanceof Integer) : (int) delayConstant : -1
```
A little less readable, but shorter

Reply:
Makes sense to do. I think you mean `?` for the first `:`, but otherwise I think this should be nicer.